File: packages/cortex/src/decay/half-lives.ts
/**
 * Half-Lives Configuration
 * 
 * Different memory types have different half-lives.
 * Core memories never decay, episodic memories decay quickly.
 */

import type { MemoryType } from '../types/index.js';

/**
 * Half-lives in days for different memory types
 */
export const HALF_LIVES: Record<MemoryType, number> = {
  core: Infinity,           // Never decays
  tribal: 365,              // Institutional knowledge is precious
  procedural: 180,          // How-to knowledge is stable
  semantic: 90,             // Consolidated knowledge persists
  episodic: 7,              // Specific interactions fade quickly
  pattern_rationale: 180,   // Pattern context is stable
  constraint_override: 90,  // Overrides need periodic review
  decision_context: 180,    // Decision context is stable
  code_smell: 90,           // Smell patterns need validation
};

/**
 * Minimum confidence before archival
 */
export const MIN_CONFIDENCE: Record<MemoryType, number> = {
  core: 0.0,                // Never archive
  tribal: 0.2,
  procedural: 0.3,
  semantic: 0.3,
  episodic: 0.1,
  pattern_rationale: 0.3,
  constraint_override: 0.2,
  decision_context: 0.3,
  code_smell: 0.2,
};

File: drift v2/crates/cortex/cortex-core/src/memory/half_lives.rs
use super::types::MemoryType;

/// Half-life in days for each memory type.
/// `None` means infinite (never decays).
pub fn half_life_days(memory_type: MemoryType) -> Option<u64> {
    match memory_type {
        // Domain-agnostic
        MemoryType::Core => None, // ∞
        MemoryType::Tribal => Some(365),
        MemoryType::Procedural => Some(180),
        MemoryType::Semantic => Some(90),
        MemoryType::Episodic => Some(7),
        MemoryType::Decision => Some(180),
        MemoryType::Insight => Some(90),
        MemoryType::Reference => Some(60),
        MemoryType::Preference => Some(120),
        // Code-specific
        MemoryType::PatternRationale => Some(180),
        MemoryType::ConstraintOverride => Some(90),
        MemoryType::DecisionContext => Some(180),
        MemoryType::CodeSmell => Some(90),
        // Universal V2
        MemoryType::AgentSpawn => Some(365),
        MemoryType::Entity => Some(180),
        MemoryType::Goal => Some(90),
        MemoryType::Feedback => Some(120),
        MemoryType::Workflow => Some(180),
        MemoryType::Conversation => Some(30),
        MemoryType::Incident => Some(365),
        MemoryType::Meeting => Some(60),
        MemoryType::Skill => Some(180),
        MemoryType::Environment => Some(90),
    }
}
